### ISSUE TITLE
Inconsistency in supported integer types on GPU

### ISSUE THREAD

**Issue by maharjun** [tensorflow/tensorflow#22249]
Please go to Stack Overflow for help and support:

https://stackoverflow.com/questions/tagged/tensorflow

If you open a GitHub issue, here is our policy:

1. It must be a bug, a feature request, or a significant problem with documentation (for small docs fixes please send a PR instead).
2. The form below must be filled out.
3. It shouldn't be a TensorBoard issue. Those go [here](https://github.com/tensorflow/tensorboard/issues).

**Here's why we have that policy**: TensorFlow developers respond to issues. We want to focus on work that benefits the whole community, e.g., fixing bugs and adding features. Support only helps individuals. GitHub also notifies thousands of people when issues are filed. We want them to see you communicating an interesting problem, rather than being redirected to Stack Overflow.

------------------------

### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**:
not relevant
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**:
Linux 4.9.0-8-amd64 #1 SMP Debian 4.9.110-3+deb9u4 (2018-08-21) x86_64 GNU/Linux
VERSION_ID="9"
VERSION="9 (stretch)"
- **Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device**:
- **TensorFlow installed from (source or binary)**:
1.10.1 (from pip binary)
- **TensorFlow version (use command below)**:
tf.VERSION = 1.10.1
tf.GIT_VERSION = v1.10.1-0-g4dcfddc5d1
tf.COMPILER_VERSION = v1.10.1-0-g4dcfddc5d1
- **Python version**:
Python 3.5.3
- **Bazel version (if compiling from source)**:
- **GCC/Compiler version (if compiling from source)**:
- **CUDA/cuDNN version**:
nvcc: NVIDIA (R) Cuda compiler driver
Copyright (c) 2005-2017 NVIDIA Corporation
Built on Fri_Nov__3_21:07:56_CDT_2017
Cuda compilation tools, release 9.1, V9.1.85

- **GPU model and memory**:
Tesla P100-PCIE-16GB

- **Exact command to reproduce**:

### Describe the problem
It appears that the kernel of `tf.reduce_sum` is not registerd for GPU's if the type of the tensor to be summed is `int64` (only registered for `tf.int32`)
Moreover the kernel of `tf.tile` is not registered for the case where the tensor to be tiled is of type `tf.int32` (only registered for `tf.int64`)

Why is there this inconsistency?

**Comment by azaks2** [comment]
tf.tile has int32 implementation (not int64)
tf.reduce has no int* implementations. There are kernels registered on GPU with int32 but they run on CPU since in most common case the tensors are small and launching GPU kernel would be inefficient.
If you indeed need an int32 op that runs on GPU you are out of luck.

Generally, there is an inconsistency since for some ops registering an int32 gpu kernel would lead to many small  ops  being launched  on GPU.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @azaks2: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.
